pkg/operator/cluster/controller.go

```diff
@@ -178,6 +201,29 @@ func (c *ClusterController) onAdd(obj interface{}) {
 func (c *ClusterController) onUpdate(oldObj, newObj interface{}) {
 	oldClust := oldObj.(*rookalpha.Cluster).DeepCopy()
 	newClust := newObj.(*rookalpha.Cluster).DeepCopy()
+	logger.Infof("update to cluster %s", newClust.Namespace)
+
+	// Check if the cluster is being deleted. This code path is called when a finalizer is specified in the crd.
+	// When a cluster is requested for deletion, K8s will only set the deletion timestamp if there are any finalizers in the list.
+	// K8s will only delete the crd and child resources when the finalizers have been removed from the crd.
+	if newClust.DeletionTimestamp != nil {
+		logger.Infof("cluster %s has a deletion timestamp", newClust.Namespace)
+		err := c.handleDelete(newClust, time.Duration(clusterDeleteRetryInterval)*time.Second)
+		if err != nil {
+			logger.Errorf("failed finalizer for cluster. %+v", err)
+			return
+		}
+		// remove the finalizer from the crd, which indicates to k8s that the resource can safely be deleted
+		c.removeFinalizer(newClust)
```

Comment:
if removing the finalizer from the CRD fails, what will cause us to retry again?

Reply:
something would have to trigger a change to the CRD. I should probably at least retry a few times before giving up since that isn't likely to be triggered